static/app/components/events/featureFlags/eventFeatureFlagList.tsx

```diff
@@ -90,34 +90,36 @@ export function EventFeatureFlagList({
     event,
   });
 
+  const suspectFlagNames: Set<string> = useMemo(() => {
+    return isSuspectError || isSuspectPending
+      ? new Set()
+      : new Set(suspectFlags.map(f => f.flag));
+  }, [isSuspectError, isSuspectPending, suspectFlags]);
+
   const hasFlagContext = Boolean(event.contexts?.flags?.values);
-  const flagValues = useMemo(() => {
-    return event.contexts?.flags?.values ?? [];
+  const eventFlags: Required<FeatureFlag>[] = useMemo(() => {
+    // At runtime there's no type guarantees on the event flags. So we have to
+    // explicitly validate against SDK developer error or user-provided contexts.
+    const rawFlags = event.contexts?.flags?.values ?? [];
+    return rawFlags.filter(
+      (f): f is Required<FeatureFlag> =>
+        f !== null &&
```

Comment:
```suggestion
        f &&
```
in case `f` is undefined

Reply:
i think we also still need the check `'flag' in f && 'result' in f` before accessing `f.flag` or `f.result` right? if the user overwrote then there's no guarantee those fields are in the object

Reply:
f.flag will be undefined if 'flag' is not in f